frontend/app/src/components/form/object-form.tsx

```diff
@@ -133,22 +133,26 @@ const NodeWithProfileForm = ({ kind, currentProfile, ...props }: ObjectFormProps
       {nodeSchema.generate_profile && (
         <ProfilesSelector
           schema={nodeSchema}
-          value={profileSelected?.display_label}
-          onChange={setProfileSelected}
+          defaultValue={currentProfiles}
+          value={selectedProfiles}
+          onChange={setSelectedProfiles}
+          currentProfiles={currentProfiles}
```

Comment:
passing currentProfiles to selectedProfiles state hook will allow you to manage everything with setSelectedProfiles